twinkle-app/twinkle.bll.services/src/test/java/com/github/donkeyrit/twinkle/bll/UserRegisterTest.java

```diff
@@ -0,0 +1,96 @@
+package com.github.donkeyrit.twinkle.bll;
+
+import com.github.donkeyrit.twinkle.bll.models.AuthenticationResult;
+import com.github.donkeyrit.twinkle.bll.security.HashManager;
+import com.github.donkeyrit.twinkle.bll.services.DefaultLoginService;
+import com.github.donkeyrit.twinkle.bll.services.contracts.LoginService;
+import com.github.donkeyrit.twinkle.dal.models.User;
+import com.github.donkeyrit.twinkle.dal.repositories.UserRepositoryImpl;
+import com.github.donkeyrit.twinkle.dal.repositories.interfaces.UserRepository;
+import jakarta.persistence.EntityManager;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+public class UserRegisterTest extends BaseTest {
+
+    private UserRepository userRepository;
+    private User user;
+
+    @BeforeClass
+    public void setUp() {
+        EntityManager session = sessionFactory.createEntityManager();
+        userRepository = new UserRepositoryImpl(session);
+    }
+
+    @Test
+    void registerNewUser(){
+        // Arrange
+        user = Utils.generateTestUser();
+        String userPassword = user.getPassword();
+        user.setPassword(HashManager.generateHash(user.getPassword()));
+
+        // Act
+        LoginService loginService = new DefaultLoginService(userRepository);
```

Comment:
Could we also create only once this service?

Reply:
Done